my-webxr-app/src/repository/DbRepository.tsx

```diff
@@ -0,0 +1,138 @@
+import {Repository} from "./Repository.tsx";
+import {DataPoint} from "./DataPoint.tsx";
+import Dexie from "dexie";
+import * as assert from "assert";
+
+// the Column class represents a column in the csv file
+// we will store csv data in the indexedDB as columns
+export class Column {
+    name: string;
+    values: Array<number | string | null>;
+
+    constructor(name: string, values: Array<number | string | null>) {
+        this.name = name;
+        this.values = values;
+    }
+}
+
+export class DbRepository extends Dexie implements Repository {
+    // Declare implicit table properties.
+    // just to inform Typescript of the object type stored in the table. Instantiated by Dexie in stores() method)
+    private columns!: Dexie.Table<Column, string>; // string = type of the primary key
+
+    constructor(dbName: string) {
+        // create a db instance
+        super(dbName);
+
+        if (this.columns) {
+            this.columns.clear();
+        }
+
+        this.version(1).stores({
+            // Declare tables, IDs and indexes
+            // set the attribute 'name' as the primary key
+            columns: 'name'
+        });
+
+        // explicitly open connection to the database
+        // if not called, db.open() will be called automatically on first query to the db
+        this.open();
+    }
+
+    /*
+    addColumn adds a column to the database
+    @param column: the column to be added to the database
+    @return Promise<void>
+     */
+    async addColumn(column: Column) {
+        return this.columns.add(column)
+            .then(() => console.log("Column " + column.name + " added to the database"))
+            .catch((err) => {
+                console.error("Error when adding column " + column.name + " to the database.");
+                throw err;
+            });
+    }
+
+    /*
+    getColumn gets a column from the database
+    @param columnName: the name of the column to be retrieved
+    @return Promise<Column>
+     */
+    private async getColumn(columnName: string) {
+        const column = await this.columns
+            .where('name')
+            .equals(columnName)
+            .toArray();
+        assert.ok(column.length <= 1, "Found more than one column with name " + columnName + "!");
+        assert.ok(column.length === 1, "Column " + columnName + " does not exist!");
+
+        return column[0];
```

Comment:
It was my intention to not catch or do anything with exceptions within the repository itself.
All repo functions return a promise. In case of an exception being thrown, the promise is rejected and the layer on top of data layer would decide what to do then.